Source/Core/Core/FifoPlayer/FifoRecorder.h

```diff
@@ -40,21 +38,21 @@ class FifoRecorder
                       u32 xfRegsSize, const u8* texMem);
 
   // Checked once per frame prior to callng EndFrame()
-  bool IsRecording() const { return m_IsRecording; }
+  bool IsRecording() const;
   static FifoRecorder& GetInstance();
 
 private:
   // Accessed from both GUI and video threads
 
+  std::recursive_mutex m_mutex;
   // True if video thread should send data
-  volatile bool m_IsRecording = false;
+  bool m_IsRecording = false;
   // True if m_IsRecording was true during last frame
-  volatile bool m_WasRecording = false;
-  volatile bool m_RequestedRecordingEnd = false;
-  volatile s32 m_RecordFramesRemaining = 0;
-  volatile CallbackFunc m_FinishedCb = nullptr;
-
-  FifoDataFile* volatile m_File = nullptr;
+  bool m_WasRecording = false;
+  bool m_RequestedRecordingEnd = false;
+  s32 m_RecordFramesRemaining = 0;
+  CallbackFunc m_FinishedCb = nullptr;
+  FifoDataFile* m_File = nullptr;
```

Comment:
If you're doing cleanup and feel like it, this can likely be made a `std::unique_ptr` now.

Reply:
Ah, that makes sense, I had wondered why it wasn't using a smart pointer.